src/axom/core/numerics/quadrature.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_NUMERICS_QUADRATURE_HPP_
+#define AXOM_NUMERICS_QUADRATURE_HPP_
+
+#include "axom/core/Array.hpp"
+
+/*!
+ * \file quadrature.hpp
+ * The functions declared in this header file find the nodes and weights of 
+ *   arbitrary order quadrature rules
+ */
+
+namespace axom
+{
+namespace numerics
+{
+
+/*!
+ * \class QuadratureRule
+ *
+ * \brief Stores a fixed array of 1D quadrature points and weights
+ */
+class QuadratureRule
+{
+  // Define friend functions so rules can only be created via compute_rule() methods
+  friend QuadratureRule compute_gauss_legendre(int npts);
+
+public:
+  QuadratureRule() = default;
+
+  //! \brief Accessor for quadrature nodes
+  double node(size_t idx) const { return m_nodes[idx]; };
+
+  //! \brief Accessor for quadrature weights
+  double weight(size_t idx) const { return m_weights[idx]; };
+
```

Comment:
I think we'll want to also expose the number of quadrature weights/nodes as a class member.

Would it make sense to also expose the order? Or does that only make sense for some quadrature rules. 
(It might be good to discuss how general we'd like this design to be beyond 1D Gauss-Legendre quadrature).

Reply:
In my opinion, the order of a quadrature rule is important, but not nearly as descriptive as the number of points. As an example, a 10th order Gauss-Legendre rule has 6 points, but a 10th order Gauss-Lobatto rule has 8 points. Since we don't currently tie the `QuadratureRule` object to a specific method, I think it's better to leave it out for now.